livelossplot/main_logger.py

```diff
@@ -28,12 +51,64 @@ def update(self, logs: dict, current_step: int or None = None) -> None:
             current_step = self.current_step
         else:
             self.current_step = current_step
-
         for k, v in logs.items():
             if k not in self.log_history:
-                self.log_history[k] = []
+                self._add_new_metric(k)
             self.log_history[k].append(LogItem(step=current_step, value=v))
 
+    def _add_new_metric(self, metric_name: str):
+        """Add empty list for a new metric and extend metric name transformations"""
+        self.log_history[metric_name] = []
+        if not self.metric_to_name.get(metric_name):
+            self._auto_generate_metrics_to_name(metric_name)
+
+    def _auto_generate_metrics_to_name(
+        self,
+        metric_name: str,
+        patterns: Tuple[Tuple[str, str]] = (
+            (r'^(?!val_).*', 'training '),
+            (r'^(val_).*', 'validation '),
+        )
+    ):
+        """
+        The function generate transforms for metric names base on patterns.
+        For example it can create transformation from val_acc to Validation Accuracy
+        :param metric_name - name of new appended metric
+        :param patterns - a tuple with pairs - pattern and value to replace it with
+        """
+        suffix = re.match(r'(.+)?(_|$)', metric_name).group(1)
```

Comment:
It always matches? Do I handle anything?

Reply:
I think it is crucial to only use regex provided. Otherwise, there is a risk of running in problems when they provide different regexes, e.g. `val-acc`, or `acc_val`, etc. It already has proved to be a problem with `lr_decay`, see e.g. #56.

Reply:
The regex idea for `'^val_'` is to match only the prefix, nothing else.

Now, as I see, the other idea is to have a single matcher for prefixes, e.g. `'^(|val_|test_).+'`.  And another field, to make it readable, e.g. `{'': 'training', 'val_': 'validation', 'test_': 'test'}` .

We can go that way or not, depending on what you consider better.